nncf/common/utils/dot_file_rw.py

```diff
@@ -36,4 +38,48 @@ def get_graph_without_data(G: nx.DiGraph) -> nx.DiGraph:
 
 
 def read_dot_graph(path: pathlib.Path) -> nx.DiGraph:
-    return nx.nx_pydot.read_dot(str(path))
+    loaded = nx.DiGraph(nx.nx_pydot.read_dot(str(path)))
+    return relabel_graph_for_dot_visualization(loaded, from_reference=True)
+
+
+def relabel_graph_for_dot_visualization(nx_graph: nx.Graph, from_reference: bool = False) -> nx.DiGraph:
+    """
+    Relabels NetworkX graph nodes to exclude reserved symbols in keys.
+        In case replaced names match for two different nodes, integer index is added to its keys.
+        While nodes keys are being updated, visualized nodes names corresponds to the original nodes names.
+
+    :param nx_graph: NetworkX graph to visualize via dot.
+    :return: NetworkX graph with reserved symbols in nodes keys replaced.
+    """
+
+    nx_graph = copy.deepcopy(nx_graph)
+
+    # .dot format reserves ':' character in node names
+    if not from_reference:
+        # dumping to disk
+        __CHARACTER_REPLACE_FROM = ":"
+        __CHARACTER_REPLACE_TO = "^"
+    else:
+        # loading from disk
+        __CHARACTER_REPLACE_FROM = "^"
+        __CHARACTER_REPLACE_TO = ":"
+
+    hits = defaultdict(lambda: 0)
+    mapping = {}
+    for original_name in nx_graph.nodes():
+        dot_name = original_name.replace(__CHARACTER_REPLACE_FROM, __CHARACTER_REPLACE_TO)
+        hits[dot_name] += 1
+        if hits[dot_name] > 1:
+            dot_name = f"{dot_name}_{hits}"
+        if original_name != dot_name:
+            mapping[original_name] = dot_name
+
+    relabeled_graph = nx.relabel_nodes(nx_graph, mapping)
+    for _, node_data in relabeled_graph.nodes(data=True):
+        if "label" in node_data:
+            node_data["label"] = node_data["label"].replace(__CHARACTER_REPLACE_FROM, __CHARACTER_REPLACE_TO)
```

Comment:
All reference dot graphs have original node names in label, this could potentially broke a name for a name with `^` simbol